model.py

```diff
@@ -197,7 +198,10 @@ def _load(path):
         assert param_state, "failed to load parameters, please check path"
 
         if self._executor is None:
-            executor = fluid.Executor(fluid.CPUPlace())._default_executor
+            # TODO: loading to CPU seems to some transform error, and only
```

Comment:
how does it behave exactly? Is it an regression in the framework code? I think it works earlier, but not completely sure.

Reply:
Run the appendix code as following:

 1. save and train (using dygraph to save before training)

```
[array([1.4861668], dtype=float32)]
[array([1.659631], dtype=float32)]
```

2. load with the original cpu executor and train （static-graph）

```
[array([1.4861668], dtype=float32)]
[array([1.6598837], dtype=float32)]
```

3. load with the gpu executor and train （static-graph）

```
[array([1.4861668], dtype=float32)]
[array([1.659631], dtype=float32)]
```

When loading with the cpu executor,  only the first step gets a right result (the results should be same with that under 1).  It also occurs on Transformer.


Appendix code:

```python
from __future__ import division
from __future__ import print_function

import contextlib
import os

import numpy as np

import paddle
from paddle import fluid
from paddle.fluid.dygraph.nn import Linear

from model import Model, CrossEntropy, shape_hints

random_seed = 123
np.random.seed(random_seed)
batch_size = 8
in_size = 32
class_num = 4

@contextlib.contextmanager
def null_guard():
    yield

guard = fluid.dygraph.guard() if True else null_guard()

class MyModel(Model):
    def __init__(self):
        super(MyModel, self).__init__()
        self._fc = Linear(in_size, class_num, act="softmax")

    def forward(self, x):
        return self._fc(x)

with guard:
    fluid.default_main_program().random_seed = random_seed
    fluid.default_startup_program().random_seed = random_seed

    model = MyModel()
    optim = fluid.optimizer.SGD(0.001, parameter_list=model.parameters())
    model.prepare(optim, CrossEntropy())
    #model.save("linear_dygraph_tmp/model")
    #model.load("linear_dygraph_tmp/model")

    for i in range(2):
        input = np.random.rand(batch_size, in_size).astype("float32")
        target = np.random.randint(0, class_num, (batch_size, 1)).astype("int64")
        out, loss = model.train(input, target, device='gpu', device_ids=[0])
        print(loss)
```

Reply:
This is caused by transfer for inplace, however it does not run as expected. 
Fix it by https://github.com/PaddlePaddle/Paddle/pull/22936